tests/drivers/gpio/gpio_basic_api/src/main.c

```diff
@@ -20,6 +20,68 @@
 #include "soc.h"
 #endif
 
+#if defined(CONFIG_GPIO_SIM)
```

Comment:
Please put all this in a separate source file that's added to `target_sources` only when this option is enabled.